docs/remote-attestation.md

```diff
@@ -1,212 +1,111 @@
-
-### Post-handshake
-
-**Client** and **Trusted Runtime** use generated _Session_ keys to exchange
-data.
+**Remote Attestation** allows a client to remotely verify the TEE hardware
+authenticity and the identity of the code running inside the TEE.
+
+If the code of the application is **reproducibly buildable**, the client can
+also check that the TEE is running the application that is expected to run.
+
+The Remote Attestation process is based on an **Attestation Report**, a data
+structure signed by the TEE platform and containing information identifying the
+code that is running inside the TEE. This report can be checked to confirm that
+it is signed by the **TEE Provider** (e.g. AMD or Intel), which results in
+evidence that the code is running on a genuine TEE platform.
 
 ## Workflow Diagram
 
 ```mermaid
 sequenceDiagram
-  participant Client
-  participant Untrusted Launcher
-  participant Trusted Runtime
-  participant TEE Platform
 
-  Note over Untrusted Launcher,TEE Platform: Load the Trusted Runtime
-
-  Untrusted Launcher->>TEE Platform: Trusted Runtime Memory Contents
-  TEE Platform -->> TEE Platform: Store a measurement of the Trusted Runtime
-  TEE Platform ->> Trusted Runtime: Start the Trusted Runtime in a secure VM
+autonumber
+
+participant C as Client
+participant U as Untrusted Launcher
+participant T as Trusted Enclave
+participant S as Secure Processor
+
+Note over U,S: Boot Trusted Enclave
+
+U->>S: Trusted Enclave Binary + firmware
+S->>T: Boot Trusted Enclave
+
+Note over U,S: Initialize
+
+U->>T: Configuration
+activate T
+T-->>T: Init logic
+T-->>T: Generate Encryption key pair
+T-->>T: Put Encryption public key hash in AttestationReport request
+T->>S: AttestationReportRequest
+S-->>S: Generate AttestationReport
+S->>T: Signed AttestationReport
+T->>U: enclave Public Key
+deactivate T
+
+Note over C,S: Fetch Enclave Public Key
+
+C->>U: GetPublicKeyRequest
+activate U
+U->>C: GetPublicKeyResponse
+deactivate U
+C-->>C: Verify Enclave Public Key attestation
+
+Note over C,S: Send Encrypted Data
+
+loop
+   C-->>C: encrypt request
+   C->>U: InvokeRequest
+   activate U
+   U->>T: InvokeRequest
+   activate T
+   T-->>T: decrypt request
+   T-->>T: process request
+   T-->>T: encrypt response
+   T->>U: InvokeResponse
+   deactivate T
+   U->>C: InvokeResponse
+   deactivate U
+   C-->>C: decrypt response
+end
+```
 
-  Trusted Runtime-->>Trusted Runtime: Generate a Signing key pair
-  Trusted Runtime-->>Trusted Runtime: Put Signing public key hash in the AttestationReport request
-  Trusted Runtime->>TEE Platform: AttestationReport request
-  TEE Platform-->>TEE Platform: Generate AttestationReport
-  TEE Platform->>Trusted Runtime: Signed AttestationReport
+### Fetch Enclave Public Key
 
-  Note over Untrusted Launcher,Trusted Runtime: Initialization
+Initially, the client connects to the untrusted launcher and requests the
+attested enclave public key. This key was generated in the enclave, but it is
```

Comment:
perhaps something like "enclave public key and supporting evidence for validating it".

Reply:
+1  for including mention of "evidence" with every mention of "enclave public key"

Reply:
Done.